packages/utils/src/db/db.ts

```diff
@@ -15,40 +15,63 @@ import {
   EthereumChallenge,
-  UINT256
-} from './types'
+  UINT256,
+  Ticket,
+  Address,
+  Hash,
+  generateChannelId
+} from '../types'
 import BN from 'bn.js'
-import { u8aEquals, u8aToNumber } from './u8a'
+import { u8aToNumber, u8aConcat, toU8a } from '../u8a'
 
 const log = debug(`hopr-core:db`)
+
 const encoder = new TextEncoder()
 const decoder = new TextDecoder()
 
-const TICKET_PREFIX = encoder.encode('tickets-')
-const SEPARATOR = encoder.encode(':')
-const PENDING_ACKNOWLEDGEMENTS_PREFIX = u8aConcat(TICKET_PREFIX, encoder.encode('pending-acknowledgement-'))
-const ACKNOWLEDGED_TICKETS_PREFIX = u8aConcat(TICKET_PREFIX, encoder.encode('acknowledged-'))
-export const pendingAcknowledgement = (halfKey: HalfKeyChallenge) => {
-  return u8aConcat(PENDING_ACKNOWLEDGEMENTS_PREFIX, halfKey.serialize())
+// key seperator:    `:`
+// key query prefix: `-`
+
+const ACCOUNT_PREFIX = encoder.encode('account-')
+const CHANNEL_PREFIX = encoder.encode('channel-')
+const COMMITMENT_PREFIX = encoder.encode('commitment-')
+const CURRENT_COMMITMENT_PREFIX = encoder.encode('commitment:current-')
+const TICKET_INDEX_PREFIX = encoder.encode('ticketIndex-')
+const PENDING_TICKETS_COUNT = encoder.encode('statistics:pending:value-')
+const ACKNOWLEDGED_TICKETS_PREFIX = encoder.encode('tickets:acknowledged-')
+const PENDING_ACKNOWLEDGEMENTS_PREFIX = encoder.encode('tickets:pending-acknowledgement-')
+
+function createChannelKey(channelId: Hash): Uint8Array {
+  return Uint8Array.from([...CHANNEL_PREFIX, ...channelId.serialize()])
+}
+function createAccountKey(address: Address): Uint8Array {
+  return Uint8Array.from([...ACCOUNT_PREFIX, ...address.serialize()])
+}
+function createCommitmentKey(channelId: Hash, iteration: number) {
+  return Uint8Array.from([...COMMITMENT_PREFIX, ...channelId.serialize(), ...toU8a(iteration, 4)])
 }
-const acknowledgedTicketKey = (challenge: EthereumChallenge, channelEpoch: UINT256) => {
-  return u8aConcat(ACKNOWLEDGED_TICKETS_PREFIX, channelEpoch.serialize(), SEPARATOR, challenge.serialize())
+function createCurrentCommitmentKey(channelId: Hash) {
+  return Uint8Array.from([...CURRENT_COMMITMENT_PREFIX, ...channelId.serialize()])
 }
-const PACKET_TAG_PREFIX: Uint8Array = encoder.encode('packets-tag-')
+function createCurrentTicketIndexKey(channelId: Hash) {
+  return Uint8Array.from([...TICKET_INDEX_PREFIX, ...channelId.serialize()])
+}
+function createPendingTicketsCountKey(address: Address) {
+  return Uint8Array.from([...PENDING_TICKETS_COUNT, ...address.serialize()])
+}
+function createAcknowledgedTicketKey(challenge: EthereumChallenge, channelEpoch: UINT256) {
+  return Uint8Array.from([...ACKNOWLEDGED_TICKETS_PREFIX, ...channelEpoch.serialize(), ...challenge.serialize()])
+}
+export const createPendingAcknowledgement = (halfKey: HalfKeyChallenge) => {
+  return Uint8Array.from([...PENDING_ACKNOWLEDGEMENTS_PREFIX, ...halfKey.serialize()])
+}
+
+const PACKET_TAG_PREFIX: Uint8Array = encoder.encode('packets-tag')
```

Comment:
All other prefixes end with a `-`, why not this one?

Reply:
it is a singleton ...

Reply:
Then its not a prefix.

Reply:
wait you're right. let me fix it properly

Reply:
changed